qendpoint-core/src/main/java/com/the_qa_company/qendpoint/core/compact/bitmap/Bitmap375Big.java

```diff
@@ -460,11 +461,67 @@ public static long binarySearch(LongArray arr, long val) {
 		return min;
 	}
 
+	public static long binarySearchNew(LongArray arr, long val) {
+		long min = 0;
+		long max = arr.length();
+		long mid;
+
+		long[] prevFound = arr.getPrevFound();
+
+		int index = (int) (val / 65536 + 1);
+
+		if (index > prevFound.length) {
+			throw new IllegalArgumentException("Index out of bounds: " + index);
+		}
+
+		if (index + 1 < prevFound.length) {
+			long t = prevFound[index + 1];
+			if (t > 0) {
+				max = Math.min(max, t);
+			}
+		}
+
+		if (index - 1 >= 0) {
+			long t = prevFound[index - 1];
+			if (t > 0) {
+				min = t;
+			}
+		}
+
+		long t = prevFound[index];
+		if (t > min && t < max) {
+			mid = t;
+		} else {
+			mid = (min + max) / 2;
+		}
+
+		while (min + 1 < max) {
+
+			long l = arr.get(mid);
+
+			if (l >= val) {
+				max = mid;
+			} else {
+				min = mid;
+			}
+			mid = (min + max) / 2;
+		}
+
+		prevFound[index] = min;
+
+		return min;
+	}
+
 	public CloseSuppressPath getBlocksPath() {
 		return blocksPath;
 	}
 
 	public CloseSuppressPath getSuperBlocksPath() {
 		return superBlocksPath;
 	}
+
+	@Override
+	public String toString() {
+		return "Bitmap375Big{}";
+	}
```

Comment:
IntelliJ likes to show the objects in the debugger, which is usually a good idea except that Bitmap375Big is usually a very very big object...so overriding toString() forces IntelliJ to use the results of that instead of analyzing and displaying the entire object.